libutils/file_lib.c

```diff
@@ -73,38 +73,28 @@ Writer *FileRead(const char *filename, size_t max_size, bool *truncated)
     return w;
 }
 
-int read_and_fill_buffer(FILE *partition, size_t bytes_expected, char *buf)
+ssize_t ReadFileStreamToBuffer(FILE *file, size_t max_bytes, char *buf)
 {
     size_t bytes_read = 0;
     size_t n = 0;
-    while (bytes_read < bytes_expected)
+    while (bytes_read < max_bytes)
     {
-        n = fread(
-            buf + bytes_read, 1, bytes_expected - bytes_read, partition);
-        if (!feof(partition) && ferror(partition))
+        n = fread(buf + bytes_read, 1, max_bytes - bytes_read, file);
+        if (ferror(file) && !feof(file))
         {
-            fprintf(
-                stderr,
-                "Error while reading from process: %s\n",
-                strerror(errno));
-            return 1;
+            return FILE_ERROR_READ;
         }
         else if (n == 0)
         {
             break;
         }
         bytes_read += n;
     }
-    if (bytes_read != bytes_expected)
```

Comment:
I guess this is intended, but now the function is not guaranteed to fill the buffer completely anymore, which was the main essence of the function in MBD.

Reply:
@oleorhagen technically, no, that was not guaranteed before, but I understand what you mean. Yes, it is intended. We just have to check the return code differently in MBD.

Reply:
I see. I guess it makes sense with the name change also.

Reply:
```C
if (ReadFile(file, bytes_expected + 1) == bytes_expected)
{
    // SUCCESS
}
else
{
    // FAIL
}
```